gk/main.c

```diff
@@ -431,8 +485,20 @@ add_ggu_policy(struct ggu_policy *policy, struct gk_instance *instance)
 		fe->u.granted.budget_byte =
 			fe->u.granted.tx_rate_kb_cycle * 1024;
 
-		/* TODO Fill up the grantor id field. */
+		/* Grantor ID comes from LPM lookup. */
+		gt_id = get_grantor_id(ltbl, &fe->flow);
+		if (gt_id < 0) {
+			RTE_LOG(ERR, GATEKEEPER,
+				"gk: initialize flow entry error in %s\n",
+				__func__);
+			fe->u.granted.grantor_id = 0;
```

Comment:
Then there's a bug here. The grantor ID isn't found and the code is assigning the first policy even if it doesn't exist.